python/cog/server/http.py

```diff
@@ -486,8 +486,6 @@ async def cancel(prediction_id: str = Path(..., title="Prediction ID")) -> Any:
         """
         Cancel a running prediction
         """
-        if not runner.is_busy():
-            return JSONResponse({}, status_code=404)
         try:
```

Comment:
404 was a weird response code to pick…

Reply:
the URL identifies a specific prediction, so if we're not running that prediction then 404 is appropriate.

There's maybe an argument that if we _did_ run that prediction but now it's completed, we could return 409.